force-app/main/default/classes/NextStepFuture.cls

```diff
@@ -58,14 +71,16 @@ public with sharing class NextStepFuture {
 
 
     // Helper method to update role name
-    private static void updateUserRoleName(UserRole[] roles){
+    @future
+    private static void updateUserRoleName(String rolesString){
         UserRole role;
-        if (roles.isEmpty()) {
+        if (String.isEmpty(rolesString) ) {
             role = new UserRole();
             role.portalType = 'None'; 
             role.CaseAccessForAccountOwner = 'Edit'; 
             // role.OpportunityAccessForAccountOwner = 'Edit'; //This may need to be uncommented depending on your orgs Organization Wide Defaults(OWD) settings. If you are receiving FIELD_INTEGRITY_EXCEPTION, field integrity exception: unknown (Opportunity access level below organization default): [unknown], uncomment this line.
         } else {
+            UserRole[] roles = (UserRole[])JSON.deserialize(rolesString, UserRole[].class);
```

Comment:
Did you note the test failure at this line of code [84]? Question you should ask yourself - if at [58], the query returns an empty list, what is the result when you serialize it at [59]? When you evaluate it at [77], will an empty list behave as expected, or will you have a side effect as a result of serializing the string. 

Is there another way to pass the data, or another place you can parse it so that your handler logic works as expected?